src/BlocksRuntime/runtime.c

```diff
@@ -32,9 +33,9 @@
 #define __has_builtin(builtin) 0
 #endif
 
-#if __has_builtin(__sync_bool_compare_and_swap)
+#if __has_builtin(__atomic_compare_exchange_n)
 #define OSAtomicCompareAndSwapInt(_Old, _New, _Ptr)                            \
-  __sync_bool_compare_and_swap(_Ptr, _Old, _New)
+  __atomic_compare_exchange_n(_Ptr, &_Old, _New, true, memory_order_release, memory_order_relaxed)
```

Comment:
If this function has the semantics of Darwin's `OSAtomicCompareAndSwapInt`, I think both memory orderings can be relaxed, but it may have to be a strong operation (it's not clear from the docs).

If you're going to use the builtin directly, you should use the builtin `__ATOMIC_RELAXED` macro.  If you're going to rely on `<stdatomic.h>`, you should use the standard operation `atomic_compare_exchange_strong_explicit`.